src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -2310,3 +2315,119 @@ def virtual_cluster_list(
 
     # List all virtual clusters in the subscription
     return client.list()
+
+###############################################
+#              sql instance failover group    #
+###############################################
+
+
+# pylint: disable=too-few-public-methods
+class InstanceFailoverPolicyType(Enum):
+    automatic = 'Automatic'
+    manual = 'Manual'
+
+
+def instance_failover_group_create(
+        cmd,
+        client,
+        resource_group_name,
+        managed_instance,
+        instance_failover_group_name,
+        partner_managed_instance_name,
+        partner_resource_group,
+        failover_policy=InstanceFailoverPolicyType.automatic.value,
+        grace_period=1):
+    '''
+    Creates a failover group.
+    '''
+
+    managed_instance_client = get_sql_managed_instances_operations(cmd.cli_ctx, None)
+    # pylint: disable=no-member
+    primary_server = managed_instance_client.get(
+        managed_instance_name=managed_instance,
+        resource_group_name=resource_group_name)
+
+    partner_server = managed_instance_client.get(
+        managed_instance_name=partner_managed_instance_name,
+        resource_group_name=partner_resource_group)
```

Comment:
Since these are existing resources, we do not allow something like `--partner-resource-group`. `--partner-managed-instance` should accept a name or ID. In either case, you shouldn't need to do a get for these resources as you can build the ARM ID very easily.

Reply:
@tjprescott  Can we force the customers to pass in the ARM resource ID only for --partner-managed-instance parameter? We would love to support the cross-subscription scenario here.

Reply:
@tjprescott 
also we would need the partner instance location information, and that is currently being retrieved with the get.

I think our options are:

1) ask customer to provider partner managed instance resource id + location 
2) ask customer to provide partner server name, resource group and subscription.

Shall we with option 1?

Reply:
3. Ask customer to provide partner managed instance name OR ID. If you need the location of the partner instance, then it would make sense to go ahead with the GET call. It looked to me like you were just doing it to get the id field.

Reply:
@tjprescott Yeah we can ask the customer to provide partner managed instance ID. However name would not be okay for cross-subscription scenarios.

Are you suggesting to ask for ID only? I realized that GET wouldn't work for cross-subscription resource either so we HAVE TO ask for the location parameter too.

I short, I suggest we go with option 1. Please let me know if that works for you.